lib/jxl/encode.cc

```diff
@@ -2327,6 +2327,87 @@ JxlEncoderStatus JxlEncoderAddImageFrame(
   return JxlErrorOrStatus::Success();
 }
 
+JxlEncoderStatus JxlEncoderAddChunkedFrame(
+    const JxlEncoderFrameSettings* frame_settings, JXL_BOOL is_last_frame,
+    JxlChunkedFrameInputSource chunked_frame_input) {
+  size_t xsize;
+  size_t ysize;
+  if (GetCurrentDimensions(frame_settings, xsize, ysize) != JXL_ENC_SUCCESS) {
+    return JXL_API_ERROR(frame_settings->enc, JXL_ENC_ERR_GENERIC,
+                         "bad dimensions");
+  }
+  // TODO(veluca): implement this without immediately buffering the whole frame.
+  JxlPixelFormat color_pixel_format;
```

Comment:
Currently we are overwriting this with when calling `get_color_channels_pixel_format` and overwrite it.

Reply:
fixed by initializing and leaving a comment, also in the analogous place below.

Reply:
yes, it will eventually get overwritten, but we have to initialize it because according to the API comment, the calllee can use its value as a hint, and without initialization that would result in an MSAN failure